src/core/api/providers/sapaicore.ts

```diff
@@ -362,6 +378,106 @@ export class SapAiCoreHandler implements ApiHandler {
 		this.options = options
 	}
 
+	/**
+	 * Checks if an error message indicates a context window issue
+	 */
+	private isContextWindowError(message: string, errorCode: number, modelName: string): boolean {
+		if (modelName.startsWith("gemini")) {
+			// Gemini returns just Internal Server Error? and no message
+			return errorCode === 500
+		}
+
+		if (errorCode !== 400) {
+			return false
+		}
+
+		const lowerMessage = message.toLowerCase()
+
+		const contextWindowPatterns = [
```

Comment:
it is a bit hacky, is it possible to add some comments on which model/provider the error belong to so it will be easier to update if there is a change from the upstream.